src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -425,12 +432,13 @@ private bool InSymbolHeader(SyntaxNode matchingNode, int position)
                 var symbolInfo = semanticModel.GetSymbolInfo((ElementAccessExpressionSyntax)originalNode, cancellationToken);
 
                 return elementAccess.WithArgumentList(
-                    UpdateArgumentList(
+                    await UpdateArgumentListAsync(
                         declarationSymbol,
                         signaturePermutation,
                         elementAccess.ArgumentList,
                         isReducedExtensionMethod: false,
-                        IsParamsArrayExpanded(semanticModel, elementAccess, symbolInfo, cancellationToken)));
+                        IsParamsArrayExpanded(semanticModel, elementAccess, symbolInfo, cancellationToken),
+                        GetSemanticModelAndRecommendations).ConfigureAwait(false));
```

Comment:
you have the semantic model above, so why do you need to grab it again and pass it through this tuple. seems odd.

Reply:
Removed indirection.